cmd/main.go

```diff
@@ -1,16 +1,80 @@
 package main
 
 import (
+	"context"
+	"flag"
 	"fmt"
+	"log"
+	"os"
+	"os/signal"
 
 	"github.com/replicatedhq/pvmigrate/pkg/migrate"
+	"github.com/replicatedhq/pvmigrate/pkg/preflight"
 	"github.com/replicatedhq/pvmigrate/pkg/version"
+	k8sclient "k8s.io/client-go/kubernetes"
 	_ "k8s.io/client-go/plugin/pkg/client/auth" // this allows accessing a larger array of cloud providers
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
 func main() {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
 	fmt.Printf("Running pvmigrate build:\n")
 	version.Print()
 
-	migrate.Cli()
+	// Cli uses CLI options to run Migrate
+	var options migrate.Options
+	var skipPreflightValidation bool
+	var preflightValidationOnly bool
+	flag.StringVar(&options.SourceSCName, "source-sc", "", "storage provider name to migrate from")
+	flag.StringVar(&options.DestSCName, "dest-sc", "", "storage provider name to migrate to")
+	flag.StringVar(&options.RsyncImage, "rsync-image", "eeacms/rsync:2.3", "the image to use to copy PVCs - must have 'rsync' on the path")
+	flag.StringVar(&options.Namespace, "namespace", "", "only migrate PVCs within this namespace")
+	flag.BoolVar(&options.SetDefaults, "set-defaults", false, "change default storage class from source to dest")
+	flag.BoolVar(&options.VerboseCopy, "verbose-copy", false, "show output from the rsync command used to copy data between PVCs")
+	flag.BoolVar(&options.SkipSourceValidation, "skip-source-validation", false, "migrate from PVCs using a particular StorageClass name, even if that StorageClass does not exist")
+	flag.IntVar(&options.PodReadyTimeout, "pod-ready-timeout", 60, "length of time to wait (in seconds) for volume validation pod(s) to go into Ready phase")
+	flag.BoolVar(&skipPreflightValidation, "skip-preflight-validation", false, "skip the volume access modes validation on the destination storage provider")
+	flag.BoolVar(&preflightValidationOnly, "preflight-validation-only", false, "skip the migration and run preflight validation only")
+
+	flag.Parse()
+
```

Comment:
I refactored this to avoid import cycles from `pvmigrate` package. Moreover, handling CLI flags in the `pvmigrate` package is not the right place to do this.